Rdmp.Core/DataViewing/ViewSupportingSqlCollection.cs

```diff
@@ -0,0 +1,58 @@
+﻿using FAnsi.Discovery.QuerySyntax;
+using Rdmp.Core.Curation.Data;
+using Rdmp.Core.Curation.Data.Dashboarding;
+using ReusableLibraryCode.DataAccess;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rdmp.Core.DataViewing
+{
+    class ViewSupportingSqlCollection : PersistableObjectCollection, IViewSQLAndResultsCollection
+    {
+
+        public SupportingSQLTable SupportingSQLTable { get => DatabaseObjects.OfType<SupportingSQLTable>().FirstOrDefault(); }
+
+        public ViewSupportingSqlCollection(SupportingSQLTable supportingSql)
+        {
+            DatabaseObjects.Add(supportingSql);
+        }
+
+        /// <summary>
+        /// Persistence constructor 
+        /// </summary>
+        public ViewSupportingSqlCollection()
+        {
+
+        }
+
+        public void AdjustAutocomplete(IAutoCompleteProvider autoComplete)
+        {
+        }
+
+        public IDataAccessPoint GetDataAccessPoint()
+        {
+            return SupportingSQLTable.ExternalDatabaseServer;
+        }
+
+        public IQuerySyntaxHelper GetQuerySyntaxHelper()
+        {
+            var syntax = SupportingSQLTable.ExternalDatabaseServer?.DatabaseType ?? FAnsi.DatabaseType.MicrosoftSQLServer;
```

Comment:
ExternalDatabaseServer=null implies the server isn't external, so assume it must be MSSQL? Does that have to be hardcoded?

Reply:
ExternalDatabaseServer is a reference to where to run the query (server,db,type and encrypted credentials - or integrated security).  If there is no server then then this functionality is not going to go well anyway but most likely it will surface as an error better elsewhere and would confuse things throwing here.